compiler/src/dmd/backend/dout.d

```diff
@@ -1005,6 +1005,8 @@ private void writefunc2(Symbol *sfunc)
     else
     {
         //printf("blockopt()\n");
+        if (f.Fflags & Finline)
+            scanForInlines(funcsym_p);  /* inline                       */
```

Comment:
Hmm, I've just realised that this is treating the pragma as `flatten` rather than `inline`. So this'll have to be ran unconditionally and the implementation itself having extra gates to ensure it doesn't try to inline every function when not optimising.

Reply:
Checked, and necessary gates are already in place.